docs/testing.md

```diff
@@ -0,0 +1,88 @@
+# Testing
+
+All testing tooling is managed by the `Makefile` on this project, therefore all you'll have to do is
+invoking one of those targets to test current changes.
+
+To run all projet tests, execute:
+
+```sh
+make test
+```
+
+## Unit-Testing
+
+To execute unit-testing present on this project, run:
+
+```sh
+make test-unit
+```
+
+## End-to-End (E2E)
+
+End-to-End tests aim to mimic the real world usage this CLI against the [Shipwright Build
+Controller][shipwrightBuild]. To run end-to-end tests make sure a newly binary is compiled (`make
```

Comment:
a newly binary is compiled -> you have compiled your latest changes to the binary